### ISSUE TITLE
Warn when shrinkwrap is overriding package.json

### ISSUE THREAD

**Issue by abrkn** [npm/npm#6781]
When running `npm install --save newdep` in the presence of a shrinkwrap, `newdep` is not installed and the user is confused.

**Comment by SlexAxton** [comment]
I *think* this is the same thing I came here to create an issue for, but wanted to make sure.

I hadn't realized that my teammate added a shrinkwrap file to our repo, and was **so confused** when I updated package.json, ran `npm install` and the wrong!!! (for some definition of wrong where I'm always right) dependency was installed instead of my new one.

Just a little warning from `npm install` that said `toplevel shinkwrap dependency mismatch with package.json - foo@1.2.4 vs. foo@1.2.5 - using foo@1.2.4 from shrinkwrap, yo.` or `\.~**whatever**~./` would be hella tite.
